vopr: add performance mode

matklad opened the pull request:
Nothing special here, just reduce most sources of randomness to get a
deterministic result.

This is not yet integrated with CFO --- there's some weirdness with client requests such that some requests are dropped even in a perfect cluster. I want to fix that before adding this to CI.

builds on top of #2817

chaitanyabhandari commented on src/vopr.zig:
```diff
@@ -237,50 +247,59 @@ pub fn main() !void {
+    if (cli_args.lite) {
+        // Don't care about convergence
+    } else {
+        simulator.transition_to_liveness_mode(.{
```
Mhmmmm I was thinking, since `options_performance` disables all faults, isn't safety mode equivalent to having a fully connected core without any perturbations? In other words, do we need liveness mode for performance testing? 🤔

matklad replied:
Ah, excellent point! So the liveness mode does two things:

* It switches the fault mode (most failures are disabled, but the remaining failures that are left become permanent).
* It also checks convergence. In the loop above, we only have checked that the cluster responded to each request, we haven't checked that for all replicas commit_min = commit_max. 

So my original thinking that we still want liveness in performance just for convergence, but of course we don't need it! The latency of requests is the only thing we care about (in the future, we might want to track "latency to convergence", but that needs to be disaggregated from the request latency). 

Rebased&simplified!

matklad committed:
vopr: add performance mode

Nothing special here, just reduce most sources of randomness to get a
deterministic result.

matklad committed:
vopr: adjust comment location

matklad committed:
vopr: hide more logs in performance mode

For performance, you are only interested in the final summary
statistics. When testing performance with non-zero percentage of faults,
theer's too much log output to scroll past.

matklad commented:
Did a couple more tweaks here after experimenting with #2821:

* reduced latencies of storage to zero (c.f. #2823)
* reduced the number of clients to 6, to be bellow client_max (7 for vopr)
* disabled the standard vopr logs, as they get in the way

chaitanyabhandari commented on src/vopr.zig:
```diff
@@ -427,6 +444,109 @@ fn options_lite(prng: *stdx.PRNG) Simulator.Options {
+        .storage = storage_options,
+        .storage_fault_atlas = storage_fault_atlas,
+        .workload = workload_options,
+        // TODO Swarm testing: Test long+few crashes and short+many crashes separately.
```
This is just a TODO for `options_swarm`, no? Or does it also apply here since we're planning on doing performance mode with faults?

matklad replied:
Good catch, removed!

chaitanyabhandari reviewed (dismissed):
LGTM! Don't want to block the PR based on a nit (https://github.com/tigerbeetle/tigerbeetle/pull/2820#discussion_r1998791614). :)
